docs/csharp/language-reference/operators/null-parameter-check.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "!! (null validation) operator - C# reference"
+description: "Learn about simplified null-parameter checks. This operator instructs the compiler to add runtime checks that the argument used for a parameter isn't null."
+ms.date: 03/30/2022
+helpviewer_keywords:
+  - "null-forgiving operator [C#]"
+  - "!! operator [C#]"
+---
+# !! (null validation) operator (C# reference)
+
+The `!!` operator, introduced in C# 11, provides null validation parameter syntax. Adding `!!` to a parameter declaration instructs the compiler to add a runtime check for that parameter. For example:
+
+``` csharp
+void Method(string name!!)
+{
+    // ...
+}
+```
+
+generates code similar to the following example:
+
+``` csharp
+void Method(string name) 
+{
+    if (name is null)
+    {
+        throw new ArgumentNullException(nameof(name));
+    }
+    // ...
+}
+```
+
+Multiple annotated parameters are checked in the order declared on the method. For example, the following code:
+
+``` csharp
+void Method(string firstName!!, string lastName!!, string? nickName, string message)
+{
+    // ...
+}
+```
+
+generates code similar to the following example:
+
+```csharp
+void Method(string firstName, string lastName, string nickName, string message) 
+{
+    if (firstName is null)
+    {
+        throw new ArgumentNullException(nameof(firstName));
+    }
+    if (lastName is null)
+    {
+        throw new ArgumentNullException(nameof(lastName));
+    }
+    // ...
+}
+```
+
+It's intended for library authors to provide runtime checks even when APIs have been annotated for nullable reference types. Some libraries support projects that aren't nullable enabled. Some projects support multiple versions including some that predate nullable reference types. These checks can simplify the necessary validation. You should only add the `!!` operator on those parameters that require a null-check for safety. In the preceding example, the `message` parameter doesn't have a runtime null check.
+
+There are a few rules that govern where you can add the `!!` operator on a parameter declaration:
+
+- The `!!` operator directs the compiler to add runtime behavior. It can't be applied to a declaration without an implementation. You can't add `!!` to the parameter of an abstract member, an interface method without an implementation, or a parameter on a delegate type declaration.
```

Comment:
> an interface method without an implementation

Does that include default interface implementations?

Reply:
It doesn't. An interface method with a default implementation has an implementation. I may try to reword this. The point is if there's implementation to add the runtime check to, the compiler can't add the runtime check.